### PR TITLE
Fix bugs and add support for the find operation

### PR DESCRIPTION
**Pull request by moshe-quantz**

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -1,68 +1,123 @@
...
-        const idx = parseInt(arrMatch[2]);
-        if (!isNaN(idx)) currval = currval[arrMatch[1]][idx];
-        else {
-          currval = currval[arrMatch[1]].map(el => {
-            return getPath(arrMatch[2], el);
-          });
-        }
+// e.g. starts with find(
+const operatorsPattern = Object.keys(operators).map(o => `^${o}\\(`).join('|');
+
```
[the remark this answers is not in the document]
Comment (moshe-quantz) [reply src/utils/index.js:49]: Changed it a little bit to ```^(${Object.keys(operators).join('|')})\((.*)\)$```

File: src/utils/index.js [hunk 2]
```diff
@@ -1,68 +1,123 @@
...
+const isOperator = str => (str.match(new RegExp(operatorsPattern)) || [])[0];
+
+module.exports.getPath = (root, path) => {
+  path = path.startsWith('.') ? path.substring(1) : path;
+
+  const openersPattern = '\\' + ['[', '(', '.'].join('|\\');
```
[the remark this answers is not in the document]
Comment (moshe-quantz) [reply src/utils/index.js]: Also removed the ```(```

File: src/utils/index.js [hunk 3]
```diff
@@ -1,68 +1,123 @@
...
-        const idx = parseInt(arrMatch[2]);
-        if (!isNaN(idx)) currval = currval[arrMatch[1]][idx];
-        else {
-          currval = currval[arrMatch[1]].map(el => {
-            return getPath(arrMatch[2], el);
-          });
-        }
+const getBracketsArg = path => {
+  const brackets = {
+    '(': 0,
```
Comment (jeeng) [inline src/utils/index.js:29]: No need to count parenthesis `(`

File: src/utils/index.js [hunk 4]
```diff
@@ -1,68 +1,123 @@
...
+
+      const {name, args} = getOperatorsArg(key);
+      if (!name) return this.getPath(root[key], nextPath);
+
+      const nextRoot = operators[name](root, ...args);
+      return this.getPath(nextRoot, nextPath);
+  }
+};
+
+const toJson = (mapping) => {
```
Comment (jeeng) [inline src/utils/index.js:89]: no need for parentheses:
`mapping => ...`

File: src/utils/index.js [hunk 5]
```diff
@@ -1,68 +1,123 @@
...
+  if (mapping.startsWith('{') && mapping.endsWith('}')) {
```
Comment (jeeng) [inline src/utils/index.js:90]: I prefer the opposite syntax:
```
if(!mapping.startsWith || !mapping.endsWith) return mapping
// rest of code
```

File: src/utils/index.js [hunk 6]
```diff
@@ -1,68 +1,123 @@
...
+  }
+
+  return mapping;
 };
 
 module.exports.transform = (root, mapping) => {
-  const transformed = {};
+  let transformed = {};
+
+  if (typeof mapping === 'string') {
```
Comment (jeeng) [inline src/utils/index.js:106]: If mapping is a string - we don't need to try and make it an object

Comment (moshe-quantz) [reply src/utils/index.js:106]: Why not? if I'm coming from a path like this: `root[map({id: 'root.id'})]` I will get the mapping ```"{id: 'root.id'}"``` (as a string)

**Pull request merged**